atlp-rwanda/phantom-fe-codebandits
# #CP-30: Feat - Simulate the bus movement

### What does this PR do?
As **a driver**, I want to be able to command the bus to start, stop, speed up, and slow down the bus so that the bus can move.

### Description of Task to be completed
- [x] The driver should be able to command the bus to start, move and stop.
- [x] The driver should be able to command the bus to speed up and slow down.
- [x] The driver should be able to update bus info on bus stops. e.g. number of passengers alighted and boarded.
- [x] The users should be able to view the bus movement in real-time.
### Demo URLs

> Frontend [https://phantom-codebantis.herokuapp.com](https://phantom-codebantis.herokuapp.com/)

> Backend [https://phantom-codebandits.herokuapp.com/](https://phantom-codebandits.herokuapp.com/)

Create an account on the backend APi and use it to login on this end
### How should this be manually tested?
- To fully test this feature you need a backend repository at your disposal.
- In the meantime us the provided links that are configured for demo purposes.

```
git clone https://github.com/atlp-rwanda/phantom-be-codebandits
cd phantom-be-codebandits
git fetch origin
git checkout ft-simulate-bus-movement-cp-30
npm install
//configure the .env according to the .env.sample file
// start the database
npm run dev
// open browser or postman (Check linked public workplace for testing)
```

### Any background context you want to provide?
As the big feature of the project. This branch will continue to receive sequential updates.
Although the PR is a work in progress will be kept open to bring attention to new ideas and contribution

### What are the relevant pivotal tracker stories?

### Screenshots (if appropriate)
